examples/testgputext.c

```diff
@@ -0,0 +1,381 @@
+#include <SDL3/SDL.h>
+#include <SDL3/SDL_main.h>
+#include <SDL3_ttf/SDL_ttf.h>
+
+#include "testgputext/shaders/spir-v.h"
+#define SDL_MATH_3D_IMPLEMENTATION
+#include "testgputext/SDL_math3d.h"
+
+#define MAX_VERTEX_COUNT 4000
+#define MAX_INDEX_COUNT  6000
+
+typedef struct Vec2
+{
+    float x, y;
+} Vec2;
+
+typedef struct Vec3
+{
+    float x, y, z;
+} Vec3;
+
+typedef struct Vertex
+{
+    Vec3 pos;
+    SDL_FColor colour;
+    Vec2 uv;
+} Vertex;
+
+typedef struct Context
+{
+    SDL_GPUDevice *device;
+    SDL_Window *window;
+    SDL_GPUGraphicsPipeline *pipeline;
+    SDL_GPUBuffer *vertex_buffer;
+    SDL_GPUBuffer *index_buffer;
+    SDL_GPUTransferBuffer *transfer_buffer;
+    SDL_GPUSampler *sampler;
+    SDL_GPUCommandBuffer *cmd_buf;
+} Context;
+
+typedef struct GeometryData
+{
+    Vertex *vertices;
+    int vertex_count;
+    int *indices;
+    int index_count;
+} GeometryData;
+
+void check_error_bool(const bool res)
+{
+    if (!res)
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "%s", SDL_GetError());
```

Comment:
Better safe then sorry. ;)
There are other instances of this.

```suggestion
    if (!res) {
        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "%s", SDL_GetError());
    }
```

Reply:
:(